[14.0] [FIX] Fields with complex context

Shide opened the pull request:
Hello @rven, @StefanRijnhart and @legalsylvain,
I've using that awesome feature for a while (pass the context used in the field), but I've found that if you click on a field with a context like this `{'default_product_id': [56]}` (note the id of the product), the `self.attrs.context` works perfectly, but if you click on a field with a conext like this one `{'default_product_id': product_id}`, throws a javascript error.
The reason is because the context goes in a direct action before being parsed.

The change I've introduced, parses the context before sending the action, so the resulting context will be the integer of the record:
`{'default_product_id': product_id}` converts to `{'default_product_id': [56]}`.

With this small change can deal with any type of context.

MT-288 @moduon

Shide commented:
The other PR that introduces the ability to add the context to the call was: https://github.com/OCA/web/pull/1961

rafaelbn reviewed (approved):
I have seen the FIX in staging

StefanRijnhart commented:
Thanks. Please check the precommit error.

Shide committed:
[14.0] [FIX] Clicking on a field with context

Clicking on a field with context like this `{'default_product_id': product_id}` doesn't throws an error.

Shide commented:
@StefanRijnhart done :)

StefanRijnhart reviewed (approved):
Thanks!

pedrobaeza commented:
/ocabot merge patch
